### PR TITLE
Add Julia implementation

### PR DESCRIPTION
**Pull request by joaoluisro**
Thought i might contribute adding the Julia implementation.

### CODE REVIEW

File: Julia/Julia.jl [hunk 1]
```diff
@@ -0,0 +1,17 @@
+longest = String[]
+max_len = 0
+invalid(word) = occursin(r"[gkmqvwxzio]", word)
```
Comment (ObserverOfTime) [inline Julia/Julia.jl]: Is this case-insensitive?

File: Julia/Julia.jl [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+for word in eachline(open("words.txt", "r"))
+    global max_len
+    if invalid(word) || max_len > length(word)
+        continue
```
Comment (ObserverOfTime) [inline Julia/Julia.jl]: Don't use `continue`.

File: .travis.yml [hunk 3]
```diff
@@ -23,12 +23,17 @@ addons:
...
+          confinement: classic
 env:
     global:
         - JAVA_HOME="/usr/lib/jvm/java-8-openjdk-amd64"
         - PERL5LIB="$HOME/perl5/lib/perl5"
         - PATH="$PATH:$HOME/dart-sdk/bin:$TRAVIS_BUILD_DIR/node_modules/.bin"
 before_install:
+    - wget https://julialang-s3.julialang.org/bin/linux/x64/1.0/julia-1.0.1-linux-x86_64.tar.gz
+    - tar xvfz julia-1.0.1-linux-x86_64.tar.gz
+    - sudo ln -s /home/ubuntu/julia-1.0.1/bin/julia /usr/local/bin/julia
```
Comment (ObserverOfTime) [inline .travis.yml]: This shouldn't be needed since the snap was already installed.

File: Julia/Julia.jl [hunk 4]
```diff
@@ -0,0 +1,16 @@
+longest = String[]
+max_len = 0
+invalid(word) = occursin(r"[gkmqvwxzioGKMQVWXZIO]", word)
```
Comment (ObserverOfTime) [inline Julia/Julia.jl]: Couldn't this be written as `r"[gkmqvwxzio]"i`?

Comment (joaoluisro) [reply Julia/Julia.jl]: Yup, definitely missed out on that one.

**Comment by ObserverOfTime** [comment]
Thank you for the contribution.

**Pull request merged**